mlflow/pipelines/steps/ingest/__init__.py

```diff
@@ -76,6 +76,7 @@ def _run(self, output_directory: str) -> BaseCard:
         _logger.debug("Successfully stored data in parquet format at '%s'", dataset_dst_path)
 
         ingested_df = read_parquet_as_pandas_df(data_parquet_path=dataset_dst_path)
+        self._validate_user_code_output(ingested_df)
```

Comment:
Instead of validating here, can we validate at the point of `load_file_as_dataframe` being called? It'd be much easier to debug in terms of stack traces and error message.